### ISSUE TITLE
False positive running nextcloud while using its exclussion rules...

### ISSUE THREAD

**Issue by feanorknd** [coreruleset/coreruleset#2973]
Hi there, lots of 921110 errors...

``
ModSecurity: Warning. Matched "Operator `Rx' with parameter `(?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+(?:\/|\w)[^\s]*(?:\s+http\/\d|[\r\n])' against variable `REQUEST_BODY' (Value: `<?xml version="1.0" ?>\x0a<d:propfind xmlns:d="DAV:">\x0a  <d:prop>\x0a    <d:getetag/>\x0a  </d:pro (23 characters omitted)' ) [file "/etc/modsecurity/owasp/coreruleset/rules/REQUEST-921-PROTOCOL-ATTACK.conf"] [line "33"] [id "921110"] [rev ""] [msg "HTTP Request Smuggling Attack"] [data "Matched Data: propfind xmlns:d="dav:">\x0a found within REQUEST_BODY: <?xml version="1.0" ?>\x0a<d:propfind xmlns:d="dav:">\x0a  <d:prop>\x0a    <d:getetag/>\x0a  </d:prop>\x0a</d:propfind>\x0a"] [severity "2"] [ver "OWASP_CRS/3.3.2"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272/220/33"] [hostname "xxxxxxxxx"] [uri "/remote.php/dav/files/a.xxxxxxxxx/"] [unique_id "166756463217.934518"] [ref "o26,25v844,105t:urlDecodeUni,t:htmlEntityDecode,t:lowercase"]
``

just detecting:

```
---fIkF7ECV---C--
<?xml version="1.0" ?>
**<d:propfind xmlns:d="DAV:">**
  <d:prop>
    <d:getetag/>
  </d:prop>
</d:propfind>`
```

Maybe could be good to include an exception for this.

Thanks.

**Comment by theseion** [comment]
Thanks for the report @feanorknd. We'll look into it.

**Comment by MrCybertux** [comment]
maybe this is relates to my issue as well? #3024

**Comment by EricDriussi** [comment]
Hi there! I think I'm having a similar issue. Only in my case it only seems to affect the connection with the desktop client.

I'll paste the log in case it helps, although I lack the knowledge to determine if it's actually the same issue or just similar.
If someone with a bigger brain wants to chime in and confirm this I can open a new issue or provide more info here :smile: 

```
2022/12/10 12:29:10 
[error] 208437#208437: *57 
[client 🬎🬎🬎🬎🬎🬎🬎🬎] ModSecurity: Access denied with code 400 (phase 2). Matched "Operator `Eq' with parameter `0' against variable `REQBODY_ERROR' (Value: `1' ) 
[file "/etc/nginx/🬎🬎🬎🬎🬎🬎🬎🬎/server-http/modsecurity-rules.conf.modsec"] 
[line "23"] 
[id "200002"] 
[rev ""] 
[msg "Failed to parse request body."] 
[data "XML parsing error: XML: Failed parsing document."] 
[severity "2"] 
[ver ""] 
[maturity "0"] 
[accuracy "0"] 
[hostname "🬎🬎🬎🬎🬎🬎🬎🬎"] 
[uri "/remote.php/dav/files/nextcloud-admin/"] 
[unique_id "167067535030.957521"] 
[ref "v830,1"], client: 🬎🬎🬎🬎🬎🬎🬎🬎, server: 🬎🬎🬎🬎🬎🬎🬎🬎, request: "PROPFIND /remote.php/dav/files/nextcloud-admin/ HTTP/2.0", host: "🬎🬎🬎🬎🬎🬎🬎🬎"
```

**Comment by theseion** [comment]
Hi @EricDriussi. Your issue is certainly similar (WebDev + PROPFIND). Still, your requests is blocked because the XML parser fails, while the original issue is that `propfind` is being detected in the request body, which is interpreted as request smuggling, since `PROPFIND` is a HTTP header in WebDav.

If you could move your question to a separate issue, that would be great.

**Comment by azurit** [comment]
Fixed with https://github.com/coreruleset/nextcloud-rule-exclusions-plugin/pull/42.
